types/nightwatch/index.d.ts

```diff
@@ -4691,3 +4691,32 @@ export interface WebDriverProtocolMobileRelated {
      */
     setContext(context: string, callback?: () => void): this;
 }
+
+/**
+ * Map of DOM element locators as shorthand string selectors based on
+ * global selector setting or ElementLocator
+ *
+ * @example
+ * const elements: PageElements {
+ * header: "h1",
+ * banner: {
+ *  locateStrategy: "css selector",
+ *  selector: "#bannerId"
+ *  }
+ * }
+ */
+export interface PageElements {
+    [key: string]: string | ElementProperties;
+}
+
+/**
+ * Type for defining page object models allowing for optional type-safe
+ * inclusion of url, elements, sections, commands, and props properties.
+ */
+export interface PageObjectModel {
+    url?: string | ((...args: any) => string);
+    elements?: PageElements;
+    sections?: EnhancedPageObjectSections;
+    commands?: any;
```

Comment:
Since commands can be `object/Array of Objects`. I think instead of `any` we can define as
```typescript
type customObject = {
  [key: string]: () => any;
}

commands?: customObject[]
```
What's your opinion on the above solution?
A more refined solution will be generic which also check types of `parameter` inside that function but we can work on it later.
eg: [Typescript Playground](https://www.typescriptlang.org/play?ssl=9&ssc=1&pln=10&pc=1#code/C4TwDgpgBAKg9geQEYCsIGNgB4YD4oC8sUEAHsBAHYAmAzlAE4QCG1clANiFANoCWlAGYQGUANIQQAGigDhogGIBXSugC6AWABQUKAH5xkkuSp0oABQZxIDUBJDbdBgN69zsyoZBqAXFGWqUAC+UH6UEABuIqFQ4VEMANza2qCQsIioGMC0AIIMDMwgOPhEzo68AJIeUADWknCCsL7pyGiYODwVarjaQUlaKeDQAKqUfOzwFZQUDLRZ45RYwyXlABTDxhQ09MyU3AarNX7DAJSE+BFwfNQxcSJnZFtmq4d+ctEVZwQXV9RnBlUwpERP1tAB6MFQAAC2QAtHwAOaUOBMQZpAKYBYAWWYYCweQKIAxwAWtBKUFGC0m0xEc0x7BwGTa2QJhXx+UKxNJuB4lCUAFskCJuqCtOh2LRgFBmByQPBWllCFB2YSuRLNqZ6AAlFhsTggVlFXYgXC4VaCFT0yi0PyGtXWr74Mo6KDi61Spi0JQcYB+Y1K5xBZIuwQoqCrN2S3iUZj8iAyC2qNRQBpQRNW2hnZ26XSe73AXmxiDJojp8pBgYupjAJQMTx5n3S+j2nF4u2WkkSnpaCvaSNSmWqjukpU8crOADkCIgwAAMlQEcAABYTvyrCjkPyShgCBGOqAb4AAOg4C+XQSk44nc2AADVmBwlBBV+HrlvgDvKAiZBEH0+wgKQoMPurhBBeV7TsAOQcBwL6rPuqw8AAjDIABMMgAMxqCcFbJsw9D9qK-YppkmD2vQRCDoU8qkcAqxUUSw4SicRESlKHBKnAtHkUekHzl+y7wVAEJQAAoqQkCYBANxIdKDAIgKVDZDISBKFKCJwFKAAM2hccyPE3vej4QOuDBPmhZwiXkClxtMKaNKk0ATh+T4TrI9DIgOtC0IiMZIKeB5wFAYAykWMz2QeQxQNeH67hOR5AA)

Reply:
I agree with going with a more specific type once we figure out the best type for this. I think we should leave as any in the iterim and then enhance later.